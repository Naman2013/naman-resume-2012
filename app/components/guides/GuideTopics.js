import React from 'react';
import PropTypes from 'prop-types';
import CenterColumn from 'components/common/CenterColumn';
import LailaTile from 'components/common/tiles/LailaTile';
import style from './GuideTopics.style';

const GuideTopics = ({ list }) => (
  <CenterColumn>
    <ul>
      {
        list.map(topic => <LailaTile key={`${topic.iconURL}-${topic.title}`} iconURL={topic.iconURL} title={topic.title} />)
      }
    </ul>
    <style jsx>{style}</style>
  </CenterColumn>
);

GuideTopics.propTypes = {
  list: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string,
    iconURL: PropTypes.string,
  })).isRequired,
};

export default GuideTopics;
