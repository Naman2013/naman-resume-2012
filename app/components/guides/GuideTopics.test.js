import React from 'react';
import { shallow } from 'enzyme';
import GuideTopics from './GuideTopics';

describe('GuideTopics', () => {
  const props = { list: [{ title: 'Saturn', iconURL: 'https://vega.slooh.com/icons/home/observatory.png' }, { title: 'Mars', iconURL: 'https://vega.slooh.com/icons/home/jupiter-icon.png' }] };
  const shallowWrapper = shallow(<GuideTopics {...props} />);
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('renders a list of `<LailaTile />`s', () => {
    expect(shallowWrapper.find('LailaTile').length).toEqual(props.list.length);
  });
});
