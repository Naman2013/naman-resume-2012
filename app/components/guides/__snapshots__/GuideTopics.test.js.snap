// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuideTopics should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GuideTopics
    list={
      Array [
        Object {
          "iconURL": "https://vega.slooh.com/icons/home/observatory.png",
          "title": "Saturn",
        },
        Object {
          "iconURL": "https://vega.slooh.com/icons/home/jupiter-icon.png",
          "title": "Mars",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ul
          className="jsx-4014916192"
        >
          <LailaTile
            iconURL="https://vega.slooh.com/icons/home/observatory.png"
            title="Saturn"
          />
          <LailaTile
            iconURL="https://vega.slooh.com/icons/home/jupiter-icon.png"
            title="Mars"
          />
        </ul>,
        <JSXStyle
          css={
            Array [
              "ul.jsx-4014916192{margin:0;padding:0;}",
            ]
          }
          styleId="4014916192"
        />,
      ],
      "theme": Object {},
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <LailaTile
              iconURL="https://vega.slooh.com/icons/home/observatory.png"
              title="Saturn"
            />,
            <LailaTile
              iconURL="https://vega.slooh.com/icons/home/jupiter-icon.png"
              title="Mars"
            />,
          ],
          "className": "jsx-4014916192",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "https://vega.slooh.com/icons/home/observatory.png-Saturn",
            "nodeType": "function",
            "props": Object {
              "iconURL": "https://vega.slooh.com/icons/home/observatory.png",
              "title": "Saturn",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "https://vega.slooh.com/icons/home/jupiter-icon.png-Mars",
            "nodeType": "function",
            "props": Object {
              "iconURL": "https://vega.slooh.com/icons/home/jupiter-icon.png",
              "title": "Mars",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            "ul.jsx-4014916192{margin:0;padding:0;}",
          ],
          "styleId": "4014916192",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ul
            className="jsx-4014916192"
          >
            <LailaTile
              iconURL="https://vega.slooh.com/icons/home/observatory.png"
              title="Saturn"
            />
            <LailaTile
              iconURL="https://vega.slooh.com/icons/home/jupiter-icon.png"
              title="Mars"
            />
          </ul>,
          <JSXStyle
            css={
              Array [
                "ul.jsx-4014916192{margin:0;padding:0;}",
              ]
            }
            styleId="4014916192"
          />,
        ],
        "theme": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <LailaTile
                iconURL="https://vega.slooh.com/icons/home/observatory.png"
                title="Saturn"
              />,
              <LailaTile
                iconURL="https://vega.slooh.com/icons/home/jupiter-icon.png"
                title="Mars"
              />,
            ],
            "className": "jsx-4014916192",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "https://vega.slooh.com/icons/home/observatory.png-Saturn",
              "nodeType": "function",
              "props": Object {
                "iconURL": "https://vega.slooh.com/icons/home/observatory.png",
                "title": "Saturn",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "https://vega.slooh.com/icons/home/jupiter-icon.png-Mars",
              "nodeType": "function",
              "props": Object {
                "iconURL": "https://vega.slooh.com/icons/home/jupiter-icon.png",
                "title": "Mars",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              "ul.jsx-4014916192{margin:0;padding:0;}",
            ],
            "styleId": "4014916192",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
